lib/SILOptimizer/Mandatory/PerformanceDiagnostics.cpp

```diff
@@ -353,7 +353,26 @@ bool PerformanceDiagnostics::visitInst(SILInstruction *inst,
   SILType impactType;
   RuntimeEffect impact = getRuntimeEffect(inst, impactType);
   LocWithParent loc(inst->getLoc().getSourceLoc(), parentLoc);
-  
+
+  if (isa<WitnessMethodInst>(inst) ||
```

Comment:
this needs to be guarded with hasFeature(Embedded), right?

Reply:
Yes you're right.